cli/src/main/java/com/devonfw/tools/ide/tool/DelegatingToolCommandlet.java

```diff
@@ -0,0 +1,25 @@
+package com.devonfw.tools.ide.tool;
+
+import java.util.Set;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.process.EnvironmentContext;
+
+public abstract class DelegatingToolCommandlet extends ToolCommandlet {
+
+  public DelegatingToolCommandlet(IdeContext context, String tool, Set<Tag> tags) {
```

Comment:
Either in constructor or via a `protected abstract` method you need to request the commandlet to delegate to.
I would even consider using a generic type `<D extends ToolCommandlet>` and then you can add:
```java
protected abstract Class<D> getDelegateClass();
```

This allows you to centralize the workaround:
```java
@Override
  public final boolean install(boolean silent, EnvironmentContext environmentContext) {
    return getCommandlet(getDelegateClass()).install(silent, environmentContext);
  }
```